kong/init.lua

```diff
@@ -1279,6 +1273,10 @@ do
       ctx.KONG_PROXY_LATENCY = ctx.KONG_RESPONSE_START - ctx.KONG_PROCESSING_START
     end
 
+    if not ctx.KONG_UPSTREAM_DNS_TIME and ctx.KONG_UPSTREAM_DNS_END and ctx.KONG_UPSTREAM_DNS_START then
+      ctx.KONG_UPSTREAM_DNS_TIME = ctx.KONG_UPSTREAM_DNS_END - ctx.KONG_UPSTREAM_DNS_START
+    end
```

Comment:
```suggestion
    else
      ctx.KONG_UPSTREAM_DNS_TIME = 0
    end
```
Let's keep the metric always present, even when we get a cache hit.